p2panda-net/src/network.rs

```diff
@@ -63,22 +66,30 @@ impl NetworkBuilder {
     ///
     /// The identifier is used during handshake and discovery protocols. Networks must use the
     /// same identifier if they wish to successfully connect and share gossip.
-    pub fn new(network_id: NetworkId) -> Self {
+    pub fn new(
+        network_id: NetworkId,
+        sync_protocol: impl for<'a> SyncProtocol<'a> + 'static,
```

Comment:
How about making setting a sync protocol optional? Not all applications need to be aware of past state

Reply:
Ah, I just implemented it myself :face_with_peeking_eye: !! Sorry...